src/gc-common.c

```diff
@@ -478,6 +477,46 @@ JL_DLLEXPORT void jl_finalize(jl_value_t *o)
     jl_finalize_th(jl_current_task, o);
 }
 
+int erase_finalizer_at(arraylist_t *list, jl_value_t *o, size_t idx)
+{
+    void **items = list->items;
+    void *v = items[idx];
+    if (o == (jl_value_t*)gc_ptr_clear_tag(v, 1)) {
+        for (size_t j = idx + 2; j < list->len; j += 2) {
+            items[j-2] = items[j];
+            items[j-1] = items[j+1];
+        }
+        list->len = list->len - 2;
```

Comment:
This also invalidates all the other `lookup_idx` values, and is potentially `O(n)` slow. We should at least only NULL the fptr and discard it on the next sweep

Reply:
Implemented.